src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1100,9 +1100,36 @@ public IRequester getRequesterBuildingForPosition(@NotNull final BlockPos pos)
 
     @Override
     @NotNull
-    public List<EntityPlayer> getMessageEntityPlayers()
+    public Set<EntityPlayer> getMessageEntityPlayers()
     {
-        return ServerUtils.getPlayersFromUUID(this.world, this.getPermissions().getMessagePlayers());
+        Set<EntityPlayer> players = new HashSet<>();
+
+        for (EntityPlayerMP player : packageManager.getSubscribers())
+        {
+            if (permissions.hasPermission(player, Action.RECEIVE_MESSAGES))
+            {
+                players.add(player);
+            }
+        }
+
+        return players;
+    }
+
+    // TODO use important message for various messages.
+    @Override
+    @NotNull
+    public Set<EntityPlayer> getImportantMessageEntityPlayers()
```

Comment:
Javadoc

Reply:
Why are they here called `Important`instead of `Global`.
Choose one and stick with it.

Reply:
because it is for important messages regarding a colony, which both the global and the close subscribers receive

Reply:
Then name that.....
Cause right now we have 
Global Subscribers
New Subscribers
ImportantMessage receivers
What else?